packages/scene-composer/src/components/panels/__tests__/__snapshots__/CommonPanelComponents.spec.tsx.snap

```diff
@@ -23,7 +23,22 @@ exports[`render correct panels. DynamicSelect should render Dynamic Select corre
 exports[`render correct panels. render ExpandableInfoSection correctly 1`] = `
 Object {
   "asFragment": [Function],
-  "baseElement": .c0 {
+  "baseElement": .c1 {
+  width: 100%;
+  display: -webkit-box;
+  display: -webkit-flex;
+  display: -ms-flexbox;
+  display: flex;
+  -webkit-flex-direction: row;
+  -ms-flex-direction: row;
+  flex-direction: row;
+  -webkit-box-pack: justify;
+  -webkit-justify-content: space-between;
+  -ms-flex-pack: justify;
+  justify-content: space-between;
+}
+
+.c0 {
   border-bottom: 1px solid colorBorderDividerDefault;
 }
```

Comment:
This is the output from taking a snapshot of styled components, and tends to be very fragile. It is also annoying to customers if they consume our components, and do tests of their own, so we should generally avoid their use entirely, and stick to classic CSS instead. (We could have a discussion with AppKit about something a bit more modern like TailwindCSS or similar, just as long as we limit the use of "css in js" style of solutions.

Reply:
we use https://www.npmjs.com/package/jest-styled-components so there should be a problem

Reply:
comparing with the classic CSS that relies on unique class name to scope style definition, I'd prefer using styled-component because we need to be super careful about the names to make them unique, otherwise it's so easy to get override by or override the other component. I've experienced with this for overlay feature where my panel container class name is the same as Grafana's panel container, then my styles are applied to Grafana panels which broke the whole Grafana dashboard.
so unless we setup something that can help solve the unique class name issue, I'd prefer using styled components

Reply:
This is easily solved by having a prefix: `tm-` on custom classnames we add. We can see even just from consuming polaris the styled-components model of doing styling is not good for library consumers and we absolutely should move away from it entirely.

I'm open to looking at other options though, scss is just sort of a default fallback. I've been playing with TailwindCSS on a personal project which may be an option.

The main reason I'm pushing on this, is it's making the problem worse by continuing to perpetuate the use of the styled components, when we should be getting rid of it. Besides the extra build step and messed up snapshots, it leads to CSS bloat, as often multiple common styles get duplicated, and the cascading effect of CSS is completely overruled. It's honestly no better than just using inline styles, and we don't need a framework for that.